contracts/wallet.sol

```diff
@@ -785,7 +784,7 @@ contract Wallet is ENSResolvable, Vault, GasTopUpLimit, LoadLimit {
         // Get the stablecoin's magnitude.
         ( ,uint256 stablecoinMagnitude, , , , , ) = _getStablecoinInfo();
         require(stablecoinMagnitude > 0, "no stablecoin");
-        _initializeLoadLimit(_DEFAULT_MAX_STABLECOIN_LOAD_LIMIT * stablecoinMagnitude);
+        _initializeLoadLimit(_MAX_STABLECOIN_LOAD_LIMIT * stablecoinMagnitude);
```

Comment:
The gas topup limit constant is called `_MAXIMUM_GAS_TOPUP_LIMIT` should we keep the prefix consistent and also call it `MAXIMUM_STABLECOIN_LOAD_LIMIT`?

Also the `MAXIMUM_GAS_TOPUP_LIMIT` is encapsulated inside the GasTopUpLimit contract.

I think it makes more sense to encapsulate `MAXIUMUM_STABLECOIN_LOAD_LIMIT` inside the LoadLimit contract since it's not used anywhere else and also you could encapsulate the `_getStablecoinInfo()` call and put it inside the LoadLimit contract. This way you won't have to do all these calculations in the wallet constructor.

Also I think the `_initializeLoadLimit` function should instead be a constructor in the LoadLimit contract which takes no parameters. Like so:

```
    constructor() internal {
        ( ,uint256 stablecoinMagnitude, , , , , ) = _getStablecoinInfo();
        maximumLoadLimit = _MAXIMUM_STABLECOIN_LOAD_LIMIT.mul(stablecoinMagnitude)
        _loadLimit = DailyLimitTrait.DailyLimit(maximumLoadLimit, maximumLoadLimit, now, 0, false);
    }
```

The wallet constructor would then look like this:
```
    constructor(...) public {
        _licenceNode = _licenceNode_;
    }
```

Reply:
I 'm fine with the renaming in order to make it consistent. The only reason we have `_DEFAULT_MAX_STABLECOIN_LOAD_LIMIT` inside the wallet is that it needs to access token-whitelist to get the stablecoin info. All the initialization (ens etc. should take place in Loadlimit instead, not sure if it is worth it.

Reply:
I think it is, otherwise it's a leaky abstraction since we expose low level details about the load limit to the wallet contract.

At the moment the TokenWhitelistable contract is initialized inside the Vault contract:
```
constructor(bytes32 _tokenWhitelistNode_ ...) TokenWhitelistable(_tokenWhitelistNode_) {
    ...
}
```
I think all we would have to change is to add this to the LoadLimit contract:

```
contract LoadLimit is TokenWhitelistable ... { 
    constructor(bytes32 _tokenWhitelistNode_ ...) TokenWhitelistable(_tokenWhitelistNode_) {
        ...
    }
    ...
}
```

Reply:
actually moving it there it is not a bad idea, you can try to see if it actually easily implementable, but branch off of this branch :)

Reply:
Could we not just do that in the next version of the contracts, it does look a lot cleaner though. Are you going to give it a go, and do we think it is going to be a big change ?

Reply:
the thing is that all these are going to be removed in the next version i.e. _one limit_, that's why i didn't want to make a lot of changes, just functional ones.

Reply:
@nostdm i have made the MAX -> MAXIMUM change. 

Also your comment on the leaky abstractions is a good one. I think what we should do, after we have merged all of this stuff in, is to rebase master onto https://github.com/tokencard/contracts/pull/520 and then we branch from that branch, which has the "one limit to rule them all" and make your proposed changes on that branch, for version 4 of the contract.

Reply:
I see, sounds good.